### PR TITLE
tengine: 2.2.3 -> 2.3.0

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update tengine to v2.3.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Izorkin** [comment]
@GrahamcOfBorg build tengine

**Comment by Izorkin** [comment]
Please help to fix build this package with sanbox mode.

**Comment by Mic92** [comment]
The buildsystem checks if /etc/resolve.conf exists.
It could be patched to skip this check to make the build work:

https://github.com/alibaba/tengine/blob/master/auto/unix#L1075

**Comment by Izorkin** [comment]
@Mic92 thanks, update.

**Comment by Izorkin** [comment]
@GrahamcOfBorg build tengine

File: pkgs/servers/http/tengine/default.nix [hunk 1]
```diff
@@ -1,27 +1,31 @@
 { stdenv, fetchurl, openssl, zlib, pcre, libxml2, libxslt
-, gd, geoip
+, gd, geoip, gperftools, jemalloc
 , withDebug ? false
 , withMail ? false
-, withIPv6 ? true
+, withSream ? false
```
Comment (c0bw3b) [inline pkgs/servers/http/tengine/default.nix]:
```suggestion
, withStream ? false
```

File: pkgs/servers/http/tengine/default.nix [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+    sha256 = "09165sdzad8bjxhnwphbags6yvxnz2rkf14p0w3vgvzssj017kqp";
   };
 
   buildInputs =
-    [ openssl zlib pcre libxml2 libxslt gd geoip ]
+    [ openssl zlib pcre libxml2 libxslt gd geoip gperftools jemalloc ]
     ++ concatMap (mod: mod.inputs or []) modules;
 
+  patches = [
+    ./check-resolf-conf.patch
```
Comment (c0bw3b) [inline pkgs/servers/http/tengine/default.nix]:
```suggestion
    ./check-resolv-conf.patch
```

File: pkgs/servers/http/tengine/default.nix [hunk 3]
```diff
@@ -54,15 +60,30 @@ stdenv.mkDerivation rec {
     "--without-mail_pop3_module"
     "--without-mail_imap_module"
     "--without-mail_smtp_module"
-  ] ++ optional withIPv6 [
-    "--with-ipv6"
+  ] ++ optional withSream [
```
Comment (c0bw3b) [inline pkgs/servers/http/tengine/default.nix]:
```suggestion
  ] ++ optional withStream [
```

File: pkgs/servers/http/tengine/default.nix [hunk 4]
```diff
@@ -54,15 +60,30 @@ stdenv.mkDerivation rec {
...
+    "--with-stream"
+    "--with-stream_ssl_module"
+    "--with-stream_realip_module"
+    "--with-stream_geoip_module"
+    "--with-stream_ssl_preread_module"
+    "--with-stream_sni"
+  ] ++ optional (withSream != true) [
```
Comment (c0bw3b) [inline pkgs/servers/http/tengine/default.nix]:
```suggestion
  ] ++ optional (withStream != true) [
```

File: pkgs/servers/http/tengine/default.nix [hunk 5]
```diff
@@ -54,15 +60,30 @@ stdenv.mkDerivation rec {
     "--without-mail_pop3_module"
     "--without-mail_imap_module"
     "--without-mail_smtp_module"
-  ] ++ optional withIPv6 [
-    "--with-ipv6"
```
Comment (c0bw3b) [inline pkgs/servers/http/tengine/default.nix]: I guess IPv6 support is now built-in by default?

Comment (Izorkin) [reply pkgs/servers/http/tengine/default.nix]: yes, ipv6 by default

**Comment by Izorkin** [comment]
thanks, update

**Comment by c0bw3b** [comment]
Result of `nix-review pr 61907` on NixOS
<details>
  <summary>1 package were build:</summary>

  - tengine
</details>

**Pull request merged**